### PR TITLE
JJ40 Refactor and Configurator support

### PR DESCRIPTION
**Pull request by noroadsleft**
`rules.mk`:

- `planck_grid` deleted from `LAYOUTS` rule

Refactor:

- `jj40.h`
  - `KEYMAP_GRID` renamed to `LAYOUT_ortho_4x12`
  - `KEYMAP_MIT` renamed to `LAYOUT_planck_mit`
  - `KEYMAP_OFFSET` renamed to `LAYOUT_offset` then corrected to `LAYOUT_offset_right` in `880f966`
    - [AliExpress page](https://www.aliexpress.com/store/product/jj40-Custom-Mechanical-Keyboard-40-PCB-programmed-40-planck-layouts-bface-firmware-gh40/3034003_32828781103.html) seems to indicate that the PCB supports an offset 2u "spacebar" in either direction, so direction of offset is specified, allowing someone who uses the other offset to add support later.
  - `LAYOUT_kc` macro retargeted
  - `KEYMAP` alias retargeted; renamed to `LAYOUT` in `880f966`
  - Aliases `LAYOUT_2U_space`, `KC_LAYOUT_ortho_4x12` and `LAYOUT_kc_ortho_4x12` deleted in `880f966`. Keymaps were refactored to use the direct matrix names, so the aliases were no longer needed.

Configurator

- Added `info.json`

### CODE REVIEW

**Comment by drashna** [comment]
@Xyverz's keymap is the issue here again. 

There are a couple solutions, but ignoring, for now.

**Comment by noroadsleft** [comment]
Re: `2d98eb0`

Forgot to check the community layouts for use of matrix aliases. Had to add the `LAYOUT_kc_ortho_4x12` alias back in.

**Comment by drashna** [comment]
For @Xyverz 
A good solution for your layout would be to replace the rgb block in your keymap with something like this: 

```c
#undef RGBLED_NUM
#ifdef KEYBOARD_planck_rev5
  #define RGBLED_NUM 10
  #define RGB_DI_PIN D1
#endif
#ifndef RGBLED_NUM 
  #define RGBLED_NUM 1
#endif
```

That or: 
```c
#ifdef KEYBOARD_planck_rev5
  #undef RGBLED_NUM
  #define RGBLED_NUM 10
  #define RGB_DI_PIN D1
#endif
```

File: keyboards/jj40/jj40.h [hunk 1]
```diff
@@ -34,61 +37,58 @@ void matrix_init_user(void);  // TODO port this to other PS2AVRGB boards
...
+  { K012, K011, K010, K09, K05, K06, K07, K08, K04, K03, K02, K01 }, \
+  { K112, K111, K110, K19, K15, K16, K17, K18, K14, K13, K12, K11 }, \
+  { XXX,  XXX,  XXX,  XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX }, \
+  { K212, K211, K210, K29, K25, K26, K27, K28, K24, K23, K22, K21 }, \
+  { K312, K311, K310, K39, K35, K3X, XXX, K38, K34, K33, K32, K31 }  \
 }
 
-#define KEYMAP_OFFSET( \
+
+#define LAYOUT_offset_right( \
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/jj40/jj40.h]: Let me tag @jackhumbert for input on this.

**Comment by Xyverz** [comment]
@drashna – Hey, thanks! I'll take a look at this. I like the bottom option there best, so I'll see if I can't use that to specify the LED configurations for my different layouts.

File: keyboards/jj40/jj40.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/jj40/jj40.h]: @noroadsleft I know you chatted with jack about this stuff.  Did you want to update this PR based on that feedback?

[the remark this answers is not in the document]
Comment (noroadsleft) [reply keyboards/jj40/jj40.h]: I think for now I will rename this matrix macro, and then if I need to make further changes I'll make another PR when the time comes.

**Comment by drashna** [comment]
@noroadsleft which layout did we want?

**Comment by drashna** [comment]
Sorry, lost track of this.  Merged as is.

**Pull request merged**